drivers/bmp180/bmp180_saul.c

```diff
@@ -42,9 +39,7 @@ static int read_pressure(void *dev, phydat_t *res)
 {
     bmp180_t *d = (bmp180_t *)dev;
 
-    bmp180_read_pressure(d, &pressure);
-
-    res->val[0] = pressure/10;
+    res->val[0] = bmp180_read_pressure(d) / 10;
```

Comment:
The unit parameter in SAUL is without decade prefix, it is the scale parameter which defines if it is hPa or Pa or kPa or whatever else.